test/std-test.cc

```diff
@@ -77,3 +77,34 @@ TEST(std_test, variant) {
   EXPECT_EQ(fmt::format("{}", v5), "variant(\"yes, this is variant\")");
 #endif
 }
+
+TEST(std_test, exception) {
+#if defined(_MSC_VER)
+#  if _MSC_VER > 1920
+  std::string str("invalid vector subscript");
+#  else
+   std::string str("invalid vector<T> subscript");
+#  endif
+#elif not defined(NDEBUG) && (defined(__clang__) || defined(__GNUC__)) || defined (__APPLE__)
+  std::string str("vector::_M_range_check: __n (which is 42) >= this->size() (which is 0)");
+#else
+    std::string str("vector");
+#endif
```

Comment:
Maybe it's easier to create a custom exception type with some predefined message instead of trying to support all compilers?

Reply:
Yeah, that's a much better idea than going down this rabbit hole!

Reply:
We can use standard exceptions.
For example (https://godbolt.org/z/qrGd455Wq):
```c++
#include <stdexcept>
#include <iostream>

int main() {
    try {
        throw std::runtime_error("My exception");
    } catch(const std::exception &e) {
        std::cerr << e.what() << std::endl; // Print: My exception
    }
    try {
        throw std::runtime_error("My exception");
    } catch(const std::runtime_error &e) {
        std::cerr << e.what() << std::endl; // Print: My exception
    }
    return 0;
}
```

Reply:
I mean, it's not only easier, but it's also the only portable way to test it IMHO